frontend/src/employee-frontend/components/unit/tab-unit-information/UnitAccessControl.tsx

```diff
@@ -68,56 +77,164 @@ const Flex = styled.div`
   justify-content: center;
 `
 
+const RowButtons = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: flex-end;
+  & > * {
+    margin-left: 10px;
+  }
+`
+
+function GroupListing({
+  unitGroups,
+  groupIds
+}: {
+  unitGroups: Record<UUID, DaycareGroupSummary>
+  groupIds: UUID[]
+}) {
+  const { i18n } = useTranslation()
+  const sortedIds = useMemo(
+    () => sortBy(groupIds, (id) => unitGroups[id]?.name),
+    [unitGroups, groupIds]
+  )
+  if (groupIds.length === 0) {
+    return <>{i18n.unit.accessControl.noGroups}</>
+  }
+  return (
+    <ExpandableList rowsToOccupy={3} i18n={i18n.common.expandableList}>
+      {[sortedIds.map((id) => <div key={id}>{unitGroups[id]?.name}</div>)]}
+    </ExpandableList>
+  )
+}
+
+function AclRowEditor({
+  row,
+  onSave,
+  onCancel,
+  unitGroups
+}: {
+  row: FormattedRow
+  onSave: (groupIds: UUID[]) => void
+  onCancel: () => void
+  unitGroups: Record<UUID, DaycareGroupSummary>
+}) {
+  const { i18n } = useTranslation()
+  const options = useMemo(
+    () => sortBy(Object.values(unitGroups), ({ name }) => name),
+    [unitGroups]
+  )
+  const [groups, setGroups] = useState<DaycareGroupSummary[]>(
+    options.filter(({ id }) => row.groupIds.includes(id))
+  )
+
+  return (
+    <Tr data-qa="acl-row">
+      <Td data-qa="name">{row.name}</Td>
+      <Td data-qa="email">{row.email}</Td>
+      <GroupMultiSelectTd>
+        <MultiSelect
+          data-qa="groups"
+          value={groups}
+          options={options}
+          getOptionId={(x) => x.id}
+          getOptionLabel={(x) => x.name}
+          onChange={(values) => setGroups(values)}
+          placeholder={`${i18n.common.select}...`}
+        />
+      </GroupMultiSelectTd>
+      <Td>
+        <RowButtons>
+          <InlineButton
+            icon={faTimes}
+            onClick={onCancel}
+            text={i18n.common.cancel}
+          />
+          <InlineButton
+            icon={faCheck}
+            data-qa="save"
+            onClick={() => onSave(groups.map(({ id }) => id))}
+            text={i18n.common.save}
+          />
+        </RowButtons>
+      </Td>
+    </Tr>
+  )
+}
+
 function AclRow({
   row,
   isDeletable,
   onClickDelete,
-  rolesAllowedToRemoveRows
+  onChangeGroups,
+  rolesAllowedToEdit,
+  unitGroups
 }: {
   row: FormattedRow
   isDeletable: boolean
   onClickDelete: () => void
-  rolesAllowedToRemoveRows?: AdRole[]
+  onChangeGroups: (ids: UUID[]) => void
+  rolesAllowedToEdit?: AdRole[]
```

Comment:
I don't like `rolesAllowedToEdit` being optional, especially when omitting the value results in everyone being able to edit.